File: 2015/Day5/Day5_part2.py
niceStrings = 0

def checkNice(string):
	if(len(string) <= 3):
		return False
	
	for i in range(1,len(string)-1):
		if(string[i-1] == string[i+1]):
			break
	else:
		return False
	
	for j in range(0,len(string)-1):
		two = string[j] + string[j+1]
		for i in range(j+2,len(string)-1):
			if(two == (string[i]+string[i+1])):
				return True

	return False

with open("2015\Day5\input.txt") as input:
    lines = input.readlines()

    for i, line in enumerate(lines):
        if checkNice(line):
            niceStrings += 1

print(niceStrings)

File: 2015/Day5/day5_part2.cpp
#include <iostream>
#include <fstream>

using namespace std;
bool checkNice(string line);

int main() {
    ifstream cin("input.txt");

    string line;
    int nice_strings = 0;

    while(!cin.eof()) {
        cin >> line;

        if (checkNice(line))
            nice_strings++;
    }

    cout << nice_strings << endl;
}

bool checkNice(string line) {
    if (line.length() <=  3)
        return false;

    bool stop = true;
    for (int i = 1, n = line.length() -1; i < n; i++) {
        if (line[i-1] == line[i+1])
            stop = false;
    }
    if (stop) return false;

    for (int j = 0; j < line.length()-1; j++) {
        string two = string() + line[j] + line[j+1];
        for (int i = j+2; i < line.length() -1; i++) 
            if (two == (string() + line[i] + line[i+1]))
                return true;
    }

    return false;
}